src/Lykke.Service.FixGateway.Services/ClientOrderIdProvider.cs

```diff
@@ -39,10 +40,14 @@ public ClientOrderIdProvider(IOperationsClient operationsClient, IConnectionMult
 
             try
             {
-                if (!await GetDatabase().SetAddAsync(_key, clientOrderId))
-                {
-                    throw new InvalidOperationException(@"Duplicate ClOrdId");
-                }
+                var db = GetDatabase();
+                var batch = db.CreateBatch();
+#pragma warning disable 4014
+                batch.KeyExpireAsync(_key, _keyExpirationPeriod);
+                batch.HashSetAsync(_key, clientOrderId, orderId.ToString());
+                batch.HashSetAsync(_key, orderId.ToString(), clientOrderId);
+#pragma warning restore 4014
+                batch.Execute();
             }
```

Comment:
I would wait for batch tasks completion. 'batch.Execute()' just fires commands, but 'batch.HashSetAsync' returns related tasks.